crates/graphql-federated-graph/src/directives/extension.rs

```diff
@@ -27,3 +27,37 @@ pub struct ExtensionDirective {
     pub name: StringId,
     pub arguments: Option<Vec<(StringId, Value)>>,
 }
+
+/// ```ignore,graphl
+/// """
+/// The directive that associates values of the `extension__Link` enum to the extension's url.
+/// """
+/// directive @extension__link(
+///   """
+///   The `@link()`ed extension's url, including name and version.
+///   """
+///   url: String!
+///   """
+///   The directives on schema definitions and extensions that are associated with the extension.
+///   """
+///   schemaDirectives: [extension__LinkSchemaDirective!]
+/// ) repeatable on ENUM_VALUE
+///
+/// input extension__LinkSchemaDirective {
+///    graph: join__Graph!
+///    name: String!
+///    arguments: DirectiveArguments
+/// }
+/// ```
+#[derive(PartialEq, PartialOrd, Clone, Debug)]
+pub struct ExtensionLinkDirective {
```

Comment:
I'm confused by the purpose of this struct, to my understanding the exact same data is in `Extension`.

Reply:
There's duplication, you're right. I had hesitations on whether it should live on `Extension` or on the enum. Let's deduplicate and move it back to extension only for now.

Reply:
After getting into the weeds of this happening in from_sdl() and render_federated_sdl(), I think it would be easier to have it _only_ on the directive, and not in the Extension struct (same thing for join__graph on join__Graph). But I've got it working now, pushing to get this PR out.

Reply:
So does that mean I'll need to find the `join__Graph` / `extension__Link` enum in federated graph to get the info for the engine Schema? Or do I miss something?

Reply:
Not now, it's only going to be in `Extension`. But yes, implementation-wise, I think it'd be cleaner if it were in the directive. To get to the info, you would have to do:

```rust
fn get_extension_url_and_schema_directives(extension: &Extension, graph: &FederatedGraph) -> (StringId, &[ExtensionLinkSchemaDirective]) {
	let enum_value = &graph[extension.enum_value_id];
   let extension_link = enum_value.directives.iter().find_map(|directive| match directive { ExtensionLinkDirective(directive) => Some(directive), _ => None).unwrap();
 Some( (exension_link.url, extension_link.schema_directives))
}
```

That's because now, in `from_sdl()`, the directives comes back as a `Directive::Other`.

Reply:
At the same time, having it in `Extension` saves you that `.unwrap()`. I just wish we had a clean, uniform way of handling `join__Graph` and `extension__Link`.

Reply:
ah don't get me wrong, nothing against your proposal. Sounds good to me if you think that's cleaner

Reply:
I'm trying to convince myself too, at the moment :p Always thinking about these data structures, it's hard to find the real sweet spots.